src/mount.c

```diff
@@ -69,11 +69,42 @@ pu_umount(const gchar *mount_point,
     return TRUE;
 }
 
+gboolean
+pu_umount_all(const gchar *device,
+              GError **error)
+{
+    g_autofree gchar *cmd_mount = g_strdup("mount");
+    g_autofree gchar *output = NULL;
+    g_autofree gchar *expr = g_strdup_printf("^%sp[1-9]+", device);
```

Comment:
As an alternative to searching for mounted partitions we could unmount everything under `/mnt/partup` with `umount /mnt/partup/p*`.

Reply:
Nice idea! However, `umount` would fail if there are no mounted partitions being left from partup or `/mnt/partup` would contain directories without any mounts in them (directories still existing after unmounting).

Reply:
Okay, I'll leave this as is for now. There is still the issue of `/dev/sd*` devices not being supported, but there are other places that need to be changed for this to work, too. This is a separate issue and should be tackled in a different pull request.